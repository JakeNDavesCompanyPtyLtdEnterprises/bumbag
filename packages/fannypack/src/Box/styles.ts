import _get from 'lodash/get';
import { css, cssClass } from '../styled';
import { breakpoint, theme } from '../utils';
import { ThemeConfig } from '../types';

export const style = styleProps => cssClass`
  & {
    ${styleProps.style};
    ${buildVisibleAttributes(styleProps)};
  }

  ${styleProps.altitude &&
    css`
      & {
        ${theme(`altitudes.${styleProps.altitude}`)(styleProps)};
      }
    `};
`;

export const Box = styleProps => cssClass`
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;

  &:focus:not(:focus-visible) {
    outline: none;
  }

  & {
    ${theme(`${styleProps.themeKey}.css.root`)(styleProps)};
  }
`;

function buildVisibleAttributes(props: { hiddenBreakpoint?: string; showBreakpoint?: string; theme: ThemeConfig }) {
  let _breakpoint = props.hiddenBreakpoint || props.showBreakpoint;
  if (!_breakpoint) return;

  return breakpoint(
    _breakpoint,
    css`
      display: none;
    `,
    { show: Boolean(props.showBreakpoint) }
  )(props);
}
