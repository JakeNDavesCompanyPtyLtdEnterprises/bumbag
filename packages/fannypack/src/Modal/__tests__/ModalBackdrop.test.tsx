import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Box } from '../../Box';
import { Modal } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ baseId: 'test' });
      return <Modal.Backdrop {...modal}>Toggle</Modal.Backdrop>;
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      function Component() {
        const modal = Modal.useState({ baseId: 'test' });
        return (
          <Modal.Backdrop use={Box} {...modal}>
            Toggle
          </Modal.Backdrop>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Modal.Backdrop props', () => {
      const { result } = renderHook(() => {
        const modal = Modal.useState({ baseId: 'test' });
        return Modal.Backdrop.useProps(modal);
      });
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      function Component() {
        const modal = Modal.useState({ baseId: 'test' });
        return (
          <Modal.Backdrop {...modal}>
            {modalBackdropProps => <Box {...modalBackdropProps}>Hello world</Box>}
          </Modal.Backdrop>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Modal.Backdrop.root should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ baseId: 'test' });
      return (
        <Modal.Backdrop overrides={{ Modal: { Backdrop: { css: { root: { backgroundColor: 'red' } } } } }} {...modal}>
          Toggle
        </Modal.Backdrop>
      );
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Modal.Backdrop.root should render correctly', () => {
    function Component() {
      const modal = Modal.useState({ baseId: 'test' });
      return <Modal.Backdrop {...modal}>Toggle</Modal.Backdrop>;
    }
    const { container } = render(<Component />, {
      theme: { Modal: { Backdrop: { css: { root: { backgroundColor: 'red' } } } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    function Component() {
      const modal = Modal.useState({ baseId: 'test' });
      return <Modal.Backdrop {...modal}>Toggle</Modal.Backdrop>;
    }
    const { container } = render(<Component />, {
      theme: { Modal: { Backdrop: { defaultProps: { className: 'red' } } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
