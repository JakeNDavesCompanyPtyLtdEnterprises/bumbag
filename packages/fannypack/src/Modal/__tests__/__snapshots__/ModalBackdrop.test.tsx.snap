// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Toggle
</div>
`;

exports[`composition hook should return with Modal.Backdrop props 1`] = `
Object {
  "className": "css-m7088q css-1pwrqjj css-3ekj6a hidden fp-Box fp-ModalBackdrop",
  "hidden": true,
  "id": undefined,
  "onAnimationEnd": [Function],
  "onTransitionEnd": [Function],
  "role": "presentation",
  "style": Object {
    "display": "none",
  },
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden red fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Toggle
</div>
`;

exports[`overrides Modal.Backdrop.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Toggle
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Toggle
</div>
`;

exports[`theming Modal.Backdrop.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19900409;
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-0 emotion-1 emotion-2 hidden fp-Box fp-ModalBackdrop"
  hidden=""
  role="presentation"
  style="display: none;"
>
  Toggle
</div>
`;
